ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -818,6 +830,7 @@ public short undelete(MessageInfo info) throws StoreException {
           IndexValue value = index.findKey(info.getStoreKey(), fileSpan,
               EnumSet.of(PersistentIndex.IndexEntryType.DELETE, PersistentIndex.IndexEntryType.UNDELETE));
           if (value != null) {
+            latestValue = value;
```

Comment:
Where is `latestValue` used? In this `if` block, no matter what the `value` is, the StoreException will be thrown and `latestValue` is never used.

Reply:
you are right, will update.